### ISSUE TITLE
Generate a subset of design principles doc for authors

### ISSUE THREAD

**Issue by LeaVerou** [w3ctag/design-principles#268]
This is a spinoff from the discussion for #177. A lot of our design principles apply to developer code as well, not just spec authors. E.g. things like JS API design and naming are more broadly useful. It may be good to annotate these principles so that in the future we can generate a version of the design principles doc targeting web developers (akin to the subset of the HTML spec aimed at web developers).

**Comment by LeaVerou** [comment]
Thankfully, Bikeshed can do that really easily: https://tabatkins.github.io/bikeshed/#conditional

**Comment by LeaVerou** [comment]
@ylafon and I started going through the principles to see what might be relevant to web developers and what is not, we recorded our work so far in [this google doc](https://docs.google.com/document/d/1EGPDc4ilTFXLLWoqyMxOJZiEpvUmS8cxyZnZy7WTats/edit?usp=sharing).

It's a shame that a lot of HTML principles that would be useful to web authors designing web components [are not there yet](https://github.com/w3ctag/design-principles/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Topic%3A+HTML%22).

**Comment by LeaVerou** [comment]
As discussed in today's breakout:
It appears there are often things brought to or discussed in the TAG that are essentially design principles for web developers exclusively, and do not apply to spec writers at all. Some examples:

- #369 
- Some of the [Guidelines for creating web platform compatible components](https://w3ctag.github.io/webcomponents-design-guidelines/#native-html-elements=)
- And several other [findings](https://tag.w3.org/findings/)

Perhaps we should have a task force with prominent web developers to come up with a design principles document that targets web developers, which will not be a subset of the design principles, but a standalone document that will pull in some of the design principles.

**Comment by LeaVerou** [comment]
Btw we started working on this with @cynthia in the July 2022 F2F. We run into some Bikeshed issues wrt conditional inclusion, but @tabatkins will fix them.

**Comment by eemeli** [comment]
Does the end result need to be a subset of the Design Principles doc, or could it include additional concerns that aren't relevant for the development of the platform itself?

In particular I'm thinking about a section like "Don't make future development of the platform more difficult", which could e.g. guide library developers away from extending the base JS `Object` and `Array` with method names that can't then easily be used when extending the JavaScript language itself.

**Comment by LeaVerou** [comment]
Yup, check out my comment above which talks about this very thing (having additional principles that are *only* present in the web developer version): https://github.com/w3ctag/design-principles/issues/268#issuecomment-1175289086 
The one you mention about built-ins is also a good candidate for that!

**Comment by torgo** [comment]
So we discussed on today's call and reiterated the consensus that this document is primarily aimed at web *platform* developers - that is, those designing new technologies for the web platform - and that some of the principles may be additionally useful for web developers. In general, principles should be considered in scope if they affect future decisions about the design of web platform features.
